config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  devise_for :users, controllers:{ 
+    registrations: 'users/registrations',
+    sessions: 'users/sessions' 
+  }
+  
+  root 'top#index'
+  resources :toilet_places do
+    resources :reviews, only: %i[new create edit update destroy]
+  end
```

Comment:
こうしないと 別の `resources :reviews` が必要になると controller の処理が混ざってしまいます。

```suggestion
  resources :toilet_places do
    resources :reviews, only: %i[new create edit update destroy], module: :toilet_places
  end
```

これが元々
- app/controllers/toilet_places_controller.rb
- app/controllers/reviews_controller.rb
- app/controllers/likes_controller.rb

module オプションをつけると
- app/controllers/toilet_places_controller.rb
- app/controllers/toilet_places/reviews_controller.rb
- app/controllers/toilet_places/reviews/likes_controller.rb
- app/controllers/reviews_controller.rb
- app/controllers/admins/reviews_controller.rb

Reply:
URL 設計の例

```
GET /toilet_places
GET /toilet_places/:id
GET /toilet_places/:id/reviews
GET /toilet_places/:id/reviews/:id

GET /toilet_places/:id/reviews/:id/likes
POST /toilet_places/:id/reviews/:id/like
DELETE /toilet_places/:id/reviews/:id/like
```

```ruby
resources :toilet_places do
  resources :reviews, only: %i[new create edit update destroy], module: :toilet_places do 
    resources :likes, only: [:index, :create, :destroy], module: :reviews
  end
end
```

参考になるブログ
https://postd.cc/how-dhh-organizes-his-rails-controllers/